java-extras/src/main/java/org/triplea/java/collections/IntegerMap.java

```diff
@@ -30,6 +30,10 @@ public IntegerMap(final IntegerMap<T> integerMap) {
     this(integerMap.mapValues);
   }
 
+  public static <X> IntegerMap<X> of(final Map<X, Integer> map) {
+    return new IntegerMap<>(map);
+  }
+
```

Comment:
What is this method for and why isn't the constructor enough?

Reply:
Convenience and avoids the diamond notation:
eg:
```
new IntegerMap<>(Map.of(...));
```
vs:
```
IntegerMap.of(Map.of(...))
```

Guava does that type of think all over the place, type inference in java is much smarter about static calls than it is about `new`

Ultimately I think `IntegerMap` is possibly a pre-Java 1.5 artifact from before there were even generics, we should consider replacing `IntegerMap` with `Map<T, Integer>`.

Reply:
@DanVanAtta I should've clarified:
I don't think I've seen this method being used inside of this PR.
So I suspect it's unused

Reply:
FWIW - test cases make use of it:
https://github.com/triplea-game/triplea/pull/6276/files#diff-98b2c58d739b161b3aa5cc607fe5b6baR42